lib/rspec/support/differ.rb

```diff
@@ -116,6 +115,12 @@ def build_hunks(actual, expected)
         HunkGenerator.new(actual, expected).hunks
       end
 
+      def final_line(last_hunk)
+        output_message = last_hunk.diff(format_type).to_s.strip
+        # Checks last char of string for + or -, appends message on match
+        output_message.gsub(/(?<=[+-]$)\z/, "\\ No newline at end of input")
```

Comment:
The approach I think I'm suggesting, is to refactor `expected_lines` and `actual_lines` to check for a missing `\n` and add the text to the array if so, then allowing the differ to proceed as normal